xonsh/xoreutils/tee.py

```diff
@@ -0,0 +1,57 @@
+import os
+import sys
+
+from xonsh.xoreutils.util import arg_handler
+
+
+def tee(args, stdin, stdout, stderr):
+    mode = 'w'
+    if '-a' in args:
+        args.remove('-a')
+        mode = 'a'
+    if '--append' in args:
+        args.remove('--append')
+        mode = 'a'
+    if '--help' in args:
+        print(HELP_STR, file=stdout)
+        return 0
+
+    errors = False
+    files = []
+    for i in args:
+        if i == '-':
+            files.append(stdout)
+        else:
+            try:
+                files.append(open(i, mode))
+            except:
+                print('tee: failed to open {}'.format(i), file=stderr)
+                errors = True
+    files.append(stdout)
+
+    while True:
+        r = stdin.read(1024)
+        if r == '':
+            break
+        for i in files:
+            i.write(r)
```

Comment:
Needs to handle the case where a pipe is closed or one of the files is on a full disk